src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoParticipantDataDaoTest.java

```diff
@@ -0,0 +1,185 @@
+import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableList;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.sagebionetworks.bridge.models.ForwardCursorPagedResourceList;
+import org.sagebionetworks.bridge.models.ParticipantData;
+import org.sagebionetworks.bridge.models.ResourceList;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.sagebionetworks.bridge.TestConstants.TEST_APP_ID;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertSame;
+
+public class DynamoParticipantDataDaoTest extends Mockito {
+
+    static final String USER_ID = "aUserId";
+    static final String IDENTIFIER = "anIdentifier";
+    static final String OFFSET_KEY = "anOffsetKey";
+    static final int PAGE_SIZE = 10;
+
+    DynamoParticipantData participantData0;
+    DynamoParticipantData participantData1;
+    DynamoParticipantData participantData2;
+    DynamoParticipantData participantData3;
+    ImmutableList<DynamoParticipantData> participantDataList;
+
+    @Mock
+    DynamoDBMapper mockMapper;
+
+    @Mock
+    PaginatedQueryList<DynamoParticipantData> mockQueryList;
+
+    @Mock
+    QueryResultPage<DynamoParticipantData> mockQueryPage;
+
+    @Captor
+    ArgumentCaptor<DynamoDBQueryExpression<DynamoParticipantData>> queryCaptor;
+
+    @Captor
+    ArgumentCaptor<ParticipantData> participantDataCaptor;
+
+    @Captor
+    ArgumentCaptor<List<DynamoParticipantData>> dataListCaptor;
+
+    @InjectMocks
+    DynamoParticipantDataDao dao;
+
+    @BeforeMethod
+    public void beforeMethod() {
+        MockitoAnnotations.initMocks(this);
+
+        // Create list with a size greater than PageSize
+        participantDataList = ImmutableList.of(
+            createParticipantData("a", "b"),
+            createParticipantData("c", "d"),
+            createParticipantData("e", "f"),
+            createParticipantData("g", "h"),
+            createParticipantData("i", "j"),
+            createParticipantData("k", "l"),
+            createParticipantData("m", "o"),
+            createParticipantData("p", "q"),
+            createParticipantData("r", "s"),
+            createParticipantData("t", "u"),
+            createParticipantData("v", "w")
+        );
+    }
+
+    @Test
+    public void testGetParticipantData() {
+        when(mockMapper.query(eq(DynamoParticipantData.class), any())).thenReturn(mockQueryList);
+        when(mockQueryList.iterator()).thenReturn(participantDataList.iterator());
```

Comment:
There's some weird stuff that goes on with PaginatedQueryList. It might be easier to use a queryHelper and spy it. See for example https://github.com/Sage-Bionetworks/BridgeServer2/blob/develop/src/main/java/org/sagebionetworks/bridge/dynamodb/DynamoHealthDataEx3Dao.java#L153 and https://github.com/Sage-Bionetworks/BridgeServer2/blob/develop/src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoHealthDataEx3DaoTest.java

Reply:
Oh, I see this as potentially the resolution to my question in your comment about using ImmutableList.copyOf(). Thank you, I'll look into this!!